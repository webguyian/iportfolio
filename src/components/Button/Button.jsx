import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Icon from 'components/Icon/Icon';
import Text from 'components/Text/Text';

const Button = forwardRef((props, ref) => {
  const { children, className, icon, size, withLabel, ...otherProps } = props;
  const baseClass = 'ui-btn';
  const iconClass = icon ? `${baseClass}--with-icon` : false;

  return (
    <button
      className={classNames(baseClass, iconClass, className)}
      ref={ref}
      {...otherProps}
    >
      {icon ? <Icon name={icon} size={size} /> : children}
      {withLabel && <Text className={`${baseClass}-label`}>{children}</Text>}
      {icon && !withLabel ? <Text type="accessible">{children}</Text> : null}
    </button>
  );
});

Button.displayName = 'Button';

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  icon: PropTypes.string,
  size: PropTypes.string,
  type: PropTypes.string,
  withLabel: PropTypes.bool
};

Button.defaultProps = {
  type: 'button',
  withLabel: false
};

export default Button;
