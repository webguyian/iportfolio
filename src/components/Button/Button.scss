.ui-btn {
  padding: 0;
  border: 0;
  background: none;
  cursor: pointer;

  &:focus {
    outline: 2px solid $blue-light;
  }

  > .ui-icon {
    pointer-events: none;
  }

  &--rounded {
    @include type-number($font-size-l);
    width: 80px;
    height: 80px;
    margin: 5px;
    padding: 14px;
    border-radius: 50%;
    background: $gray-medium;
  }

  &--with-icon {
    padding: 2px;
  }

  &-label {
    margin-left: 10px;
  }
}
