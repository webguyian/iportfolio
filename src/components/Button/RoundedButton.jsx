import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Button from 'components/Button/Button';

const RoundedButton = props => {
  const baseClass = 'ui-btn--rounded';

  return (
    <Button {...props} className={classNames(baseClass, props.className)} />
  );
};

RoundedButton.propTypes = {
  className: PropTypes.string
};

export default RoundedButton;
