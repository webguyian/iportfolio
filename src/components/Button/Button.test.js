import React from 'react';
import { create } from 'react-test-renderer';

import Button from './Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const component = create(<Button />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with children', () => {
    const component = create(<Button>Button text</Button>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with icon', () => {
    const component = create(<Button icon="plus-circle">Add</Button>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with icon and label', () => {
    const component = create(
      <Button icon="plus-circle" withLabel>
        Add
      </Button>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom className', () => {
    const component = create(<Button className="custom-btn" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
