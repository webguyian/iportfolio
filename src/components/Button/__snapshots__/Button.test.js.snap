// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="ui-btn"
  type="button"
/>
`;

exports[`<Button /> renders correctly with children 1`] = `
<button
  className="ui-btn"
  type="button"
>
  Button text
</button>
`;

exports[`<Button /> renders correctly with custom className 1`] = `
<button
  className="ui-btn custom-btn"
  type="button"
/>
`;

exports[`<Button /> renders correctly with icon 1`] = `
<button
  className="ui-btn ui-btn--with-icon"
  type="button"
>
  <i
    className="ui-icon ui-icon-plus-circle fas fa-plus-circle"
  />
  <span
    className="ui-text ui-text--accessible"
  >
    Add
  </span>
</button>
`;

exports[`<Button /> renders correctly with icon and label 1`] = `
<button
  className="ui-btn ui-btn--with-icon"
  type="button"
>
  <i
    className="ui-icon ui-icon-plus-circle fas fa-plus-circle"
  />
  <span
    className="ui-text ui-text--body ui-btn-label"
  >
    Add
  </span>
</button>
`;
